Signup block

sneharora1 opened the pull request:
Sign up block

Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #60 

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/draft/sneh/equity
- After: https://signup--icicidirect--aemsites.hlx.live/draft/sneh/equity

- [x] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
signup block | [doc-link](tbd)

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+.block.sign-up .signup-container {
+    box-shadow: 0 1px 3px 0 var(--border-color-dark-grey);
```
use global variable `--space-`

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+     border: 1px solid var(--background-color-grey);
+     border-radius: var(--space-24px);
+     background-color: var(--color-white);
+     box-shadow: 0 20px 60px 20px var( --color-darker), 0 2px 4px 0 var(--color-yellowish-white);
```
use `rem` and global variable

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+     transform: translateY(-50%);
+ }
+ 
+
```
remove extra lines

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+    display: inline-block;
+    margin-bottom: 0.5rem;
+}
+
```
remove extra lines

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+ .block.sign-up .signup-container .signup-form-group label::before {
+     content: "";
+     position: absolute;
+     bottom: 5px;
```
global variable

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+     bottom: 5px;
+     left: 50%;
+     transform: translateX(-50%);
+     width: 100px;
```
`rem`

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+     left: 50%;
+     transform: translateX(-50%);
+     width: 100px;
+     height: 3px;
```
global

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+ }
+
+ .block.sign-up .text-left {
+     text-align: left !important;
```
avoid `!important`

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+ }
+  
+ .block.sign-up .promotional-text {
+     margin-top: 0.25rem !important;
```
avoid `!important`

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+ .block.sign-up .promotional-text {
+     margin-top: 0.25rem !important;
+     color:var(--color-darker-gray);
+     display: block !important;
```
avoid `!important`

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+
+
+ .block.sign-up .justify-content-center {
+     justify-content: center !important;
```
avoid `!important`

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+    display: block;
+    width: 100%;   
+    font-size: 1rem;
+    border: 1px solid var( --border-color-lighter-gray);
```
global

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+.block.sign-up input, .block.sign-up .phonenumber-textbox::placeholder {
+    font-size: 1rem;
+}
+
```
extra line

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.js:
```diff
@@ -0,0 +1,148 @@
+  submitButton.textContent = buttontitle;
+  submitButton.addEventListener('click', handleOpenAccountSubmit);
+  return submitButton;
+}
```
add one empty line

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.js:
```diff
@@ -0,0 +1,148 @@
+  spanElement.textContent = errormessage;
+  return spanElement;
+}
+function createTitle(titleHTML) {
```
add one empty line

sneharora1 replied:
fixed

ravverma commented on styles/fonts.css:
```diff
@@ -164,4 +182,5 @@
-}
\ No newline at end of file
+}
+
```
extra line

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+    }
+} 
+
+@media (width < 48rem) {
```
same as above

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,216 @@
+}
+
+
+@media (width <= 64rem) {
```
We are developing mobile first so the default css should mobile specific and media queries for other screens , so the width should with `>=`

sneharora1 replied:
fixed

ravverma commented on blocks/sign-up/sign-up.js:
```diff
@@ -0,0 +1,148 @@
+ * @param {string} key - The key to search for within the block.
+ * @returns {HTMLElement|string} - The found HTML element or an empty string if no match is found.
+ */
+function findHTMLElementFromBock(block, key) {
```
I believe this methods is using for fetching elements, we should avoid iteration overall block to find desired element.
Instead use `.getElementById` and `.querySelector` , And I hope it will improve the performance as well.

sneharora1 replied:
Changed to querySelector as recommended but this will fixed order of properties in the table.

anagarwa commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,214 @@
+    padding: 0 var(--space-15px);
+    max-width: 81.25rem;
+    margin: 0 auto;
+    box-shadow: 0 0 7px -1px var(--color-darkest-gray);
```
global varriables

anagarwa commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,214 @@
+
+ .block.sign-up .signup-container .signup-form-group {
+     text-align: center;
+     border: 1px solid var(--background-color-grey);
```
global varriable

sneharora1 replied:
fixed

anagarwa commented on styles/styles.css:
```diff
@@ -172,6 +183,12 @@
   --space-13px: 0.813rem;
   --space-14px: 0.875rem;
   --space-15px: 0.938rem;  
+  --space-20px: 1.25rem; 
```
AFAIK, this we are keeping for less than 1 rem only

sdmcraft reviewed (commented):
This doesn't have `reCaptcha` . You can consider reusing the [forms block](https://www.aem.live/developer/forms) which has `reCaptcha` support but you'll have to style it as per your needs.

JiangLong2019 commented:
The reCaptcha works on the page now, based on the discussion with Dinesh maybe we can mock a API 

<img width="1020" alt="image" src="https://github.com/aemsites/icicidirect/assets/57093988/26c29dd0-aa91-44cf-b988-049252c429b4">

vivesing commented:
@sneharora1 , please include rest of the merged blocks in your PR doc https://signup--icicidirect--aemsites.hlx.page/draft/sneh/signup

vivesing commented on blocks/sign-up/sign-up.js:
```diff
@@ -0,0 +1,130 @@
+  const {
+    promotionaltext, placeholdertext, buttontitle, errormessage,
+  } = blockConfig;
+  const titleHTML = block.querySelectorAll(':scope > div')[0].children[1];
```
Would it make sense to make it a little resilient by iterating over all children and looking for children[0] to match `title` as value? We can do it here itself in this block code.

anagarwa replied:
@vivesing you are right , but we decided to update readblock config to have a default class based on key to div.
that has been tracked https://github.com/aemsites/icicidirect/issues/119. 
I will create a PR for the same. Till then we can keep this implementation as done in another blocks as well.

vivesing commented:
Original Site have an orange bar below the title, thats missing in the PR link
![image](https://github.com/aemsites/icicidirect/assets/38683124/598439f7-f66d-4059-b193-a50b996ac233)

vivesing commented:
Some of the title is non-bold in the original site, in mobile view
Original: 
![image](https://github.com/aemsites/icicidirect/assets/38683124/423db6b3-9118-4a89-a558-cebfb13e67d6)
PR: 
![image](https://github.com/aemsites/icicidirect/assets/38683124/74dd1a91-b123-4fe4-b069-a6d25453ffee)

sneharora1 commented:
> This doesn't have `reCaptcha` . You can consider reusing the [forms block](https://www.aem.live/developer/forms) which has `reCaptcha` support but you'll have to style it as per your needs.

Added captcha

> The reCaptcha works on the page now, based on the discussion with Dinesh maybe we can mock a API
> 
> <img alt="image" width="1020" src="https://private-user-images.githubusercontent.com/57093988/320698360-26c29dd0-aa91-44cf-b988-049252c429b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2Nzg0MDMsIm5iZiI6MTcxMjY3ODEwMywicGF0aCI6Ii81NzA5Mzk4OC8zMjA2OTgzNjAtMjZjMjlkZDAtYWE5MS00NGNmLWI5ODgtMDQ5MjUyYzQyOWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDE1NTUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhMTM4ZGZkNTA0ODY1ZDhhMzQ4MWMxNjMxZGVkNzcxYjNjZjVlYjk2ZTUzOGMyYTMxMzY5MDFmNjQ5ZmEzN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Gio_drzCd0RO1ErSsXMRcTuB5Eo3i8_-3aXGqkXCh08">

Added

sneharora1 commented:
> @sneharora1 , please include rest of the merged blocks in your PR doc https://signup--icicidirect--aemsites.hlx.page/draft/sneh/signup

Done

sneharora1 commented:
> Original Site have an orange bar below the title, thats missing in the PR link ![image](https://private-user-images.githubusercontent.com/38683124/320810189-598439f7-f66d-4059-b193-a50b996ac233.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2ODIwNjUsIm5iZiI6MTcxMjY4MTc2NSwicGF0aCI6Ii8zODY4MzEyNC8zMjA4MTAxODktNTk4NDM5ZjctZjY2ZC00MDU5LWIxOTMtYTUwYjk5NmFjMjMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDE2NTYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE5YWQxNjdiYThiZDAxYzVkODcyMmU3MjE0ZmFhOGVhZGJhYjE2ZWYwZTU0YWI5Y2U1ZDJlOTI3NDE2ZGUwNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._n20FPitzuER6S9zioLgXROw4Mpmwx6BeyhMPUBaTUI)

Thanks for catching this..the css somehow got lost..It was there earlier :). Fixed it.

sneharora1 commented:
> Some of the title is non-bold in the original site, in mobile view Original: ![image](https://private-user-images.githubusercontent.com/38683124/320810854-423db6b3-9118-4a89-a558-cebfb13e67d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2ODIwNjUsIm5iZiI6MTcxMjY4MTc2NSwicGF0aCI6Ii8zODY4MzEyNC8zMjA4MTA4NTQtNDIzZGI2YjMtOTExOC00YTg5LWE1NTgtY2ViZmIxM2U2N2Q2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDE2NTYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4YmM0M2ZjMzdmMDFmZDE4YWQ0MDgwZGU3NmZkNzNhYWNmMTA0NGYxZjdkMTZkNTMyNjE2MjM2OTE2NjEyNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XLJeFWP4htPGAqH1OQXueNI9W55jChbXJ3EUUmcJ6iA) PR: ![image](https://private-user-images.githubusercontent.com/38683124/320810963-74dd1a91-b123-4fe4-b069-a6d25453ffee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2ODIwNjUsIm5iZiI6MTcxMjY4MTc2NSwicGF0aCI6Ii8zODY4MzEyNC8zMjA4MTA5NjMtNzRkZDFhOTEtYjEyMy00ZmU0LWIwNjktYTZkMjU0NTNmZmVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDE2NTYwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhMjZjNzIwZDMxNzQxMGNlN2E3YzcyY2NiMGFlNDU3ZmFmM2I5ZmYzODRlZDAyYzI0N2VhNjc4NTVmNTJhMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Niu4hCgvOiC8ZBvVTvHNHleVtoM8S0uNykCP6LjJbcM)

Fixed it.

vivesing commented:
@sneharora1 , one minor bug, the captacha is not aligned in smaller width
![image](https://github.com/aemsites/icicidirect/assets/38683124/6ac20328-2f85-43f9-a028-9dd326de2e9a)

ravverma commented on blocks/sign-up/sign-up.css:
```diff
@@ -0,0 +1,227 @@
+        max-width: 26.25rem;
+    }
+
+    .block.sign-up .signup-container {
```
It is duplicate as in mobile view has already similar style.

ravverma commented on styles/styles.css:
```diff
@@ -53,6 +58,7 @@
   --text-color-light: #4a4a4a;
   --text-color-blue: #043C6D;
   --text-color-gray: #9B9B9B;
+  --text-color-light-gray: rgb(14 31 53 / 50%);
```
convert to hex

ravverma replied:
I have checked no diff in using `#0e1f3580`

vivesing replied:
@sneharora1 we can use the nearby grey color for example: existing ` --color-light-gray-dark`

sneharora1 commented:
> @sneharora1 , one minor bug, the captacha is not aligned in smaller width ![image](https://private-user-images.githubusercontent.com/38683124/321107601-6ac20328-2f85-43f9-a028-9dd326de2e9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3NDIxMjQsIm5iZiI6MTcxMjc0MTgyNCwicGF0aCI6Ii8zODY4MzEyNC8zMjExMDc2MDEtNmFjMjAzMjgtMmY4NS00M2Y5LWEwMjgtOWRkMzI2ZGUyZTlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDEwVDA5MzcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5YTNhZDI1NjdmNmYxODg0MmZkYmUwYjQ3NjQ4MThkMzM2OTZlODVkYTRmM2QwM2I0YzI1YzQ4ZTRmMWM1Y2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4x3fAmmRhKzRlpl8gS3bDL3W2pwZh7xh2EDj48UpuHU)

I have fixed it for sizes>375px No  easy way to fix since the styling for captcha is coming from google.
